testsuite/tests/typing-poly/poly.ml

```diff
@@ -1516,30 +1516,33 @@ val n : < m : 'x 'a. ([< `Foo of 'x ] as 'a) -> 'x > = <obj>
 let n =
   object method m : 'x. [< `Foo of 'x] -> 'x = fun x -> assert false end;;
 [%%expect {|
-val n : < m : 'x. [< `Foo of 'x ] -> 'x > = <obj>
+Line 2, characters 9-68:
+2 |   object method m : 'x. [< `Foo of 'x] -> 'x = fun x -> assert false end;;
+             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The method m has type 'a -> 'b but is expected to have type
+         [< `Foo of 'x ] -> 'c
+       The universal variable 'x would escape its scope
```

Comment:
Note this example is still marked as `ok` even if it fails now.

This seems like a slight regression, however the slightly modified version 
```ocaml
object method m: 'x. [< `Foo of 'x] -> 'x = fun (`Foo x) -> x end
```
was already failing, thus this seems like a pre-existing issue.

Reply:
Indeed, this code is basically wrong. A univar should not appear in a type whose row variable is bound at a more external position. This is correctly enforced when the row variable is also a univar, but not when it is a normal variable, ending up accepting meaningless types that cannot really be used (trying to call this method would wail, IIRC).